### PR TITLE
Update and add test case for exception

### PR DESCRIPTION
**Pull request by GinnyJI**
Tested: make test

### CODE REVIEW

**Comment by GinnyJI** [comment]
tracked at #1123

File: incubator/hnc/internal/reconcilers/object_test.go [hunk 1]
```diff
@@ -45,30 +45,46 @@ var _ = Describe("Secret", func() {
...
+		setParent(ctx, barName, fooName)
+		setParent(ctx, bazName, fooName)
+
+		makeObject(ctx, "Role", fooName, "testrole")
+		Eventually(hasObject(ctx, "Role", barName, "testrole")).Should(BeTrue())
+		Eventually(hasObject(ctx, "Role", bazName, "testrole")).Should(BeTrue())
+
+		// Update the secret so that it does NOT propogate to bazName
+		a := map[string]string{"propagate.hnc.x-k8s.io/select": "!" + bazName + ".tree.hnc.x-k8s.io/depth"}
+		Expect(updateObjectWithAnnotation(ctx, "Role", fooName, "testrole", a)).Should(BeNil())
```
Comment (adrianludwin) [inline incubator/hnc/internal/reconcilers/object_test.go:84]: You can replace `BeNil()` with `Succeed()` here - same effect, but slightly easier to read.

File: incubator/hnc/internal/reconcilers/test_helpers_test.go [hunk 2]
```diff
@@ -195,6 +195,21 @@ func makeObjectWithAnnotation(ctx context.Context, kind string, nsName,
 	createdObjects = append(createdObjects, inst)
 }
 
+// updateObjectWithAnnotation gets an object given it's kind, nsName and name, adds the annotation
+// and updates this object
+func updateObjectWithAnnotation(ctx context.Context, kind string, nsName,
```
Comment (adrianludwin) [inline incubator/hnc/internal/reconcilers/test_helpers_test.go:200]: Does it make more sense to have `makeObject` return an `*unstructured.Unstructured` and then you can just call `inst.SetAnnotations()` and `k8sClient.Update()` directly in the test function? I don't want to have _too_ many special-purpose helper functions.

**Review by adrianludwin — COMMENTED** [review]
lgtm with two changes for you to consider. LMK if you'd rather submit this as-is.

File: incubator/hnc/internal/reconcilers/object_test.go [hunk 3]
```diff
@@ -45,30 +45,46 @@ var _ = Describe("Secret", func() {
...
+
+	PIt("should remove object in selected namespace when selector is updated", func() {
+		setParent(ctx, barName, fooName)
+		setParent(ctx, bazName, fooName)
+
+		makeObject(ctx, "Role", fooName, "testrole")
+		Eventually(hasObject(ctx, "Role", barName, "testrole")).Should(BeTrue())
+		Eventually(hasObject(ctx, "Role", bazName, "testrole")).Should(BeTrue())
+
+		// Update the secret so that it does NOT propogate to bazName
```
Comment (yiqigao217) [inline incubator/hnc/internal/reconcilers/object_test.go:82]: r/secret/role now. Same to other comments.

**Review by yiqigao217 — COMMENTED** [review]
Just confirming that this is only tests updates and no reconciler behavior changes, right? What is the new PR for https://github.com/kubernetes-sigs/multi-tenancy/pull/1122#issuecomment-698935005?

File: incubator/hnc/internal/reconcilers/object_test.go (hunk 3 above)
Comment (GinnyJI) [reply incubator/hnc/internal/reconcilers/object_test.go:82]: Done

File: incubator/hnc/internal/reconcilers/object_test.go (hunk 1 above)
Comment (GinnyJI) [reply incubator/hnc/internal/reconcilers/object_test.go:84]: Done

File: incubator/hnc/internal/reconcilers/object_test.go [hunk 4]
```diff
@@ -45,30 +45,46 @@ var _ = Describe("Secret", func() {
...
-		Consistently(hasObject(ctx, "Secret", bazName, "fooSecret")).Should(BeFalse())
+		Eventually(hasObject(ctx, "Role", barName, "testrole")).Should(BeTrue())
+		Consistently(hasObject(ctx, "Role", bazName, "testrole")).Should(BeFalse())
 	})
 
 	PIt("should propagate object only to selected namespace using select", func() {
 		setParent(ctx, barName, fooName)
 		setParent(ctx, bazName, fooName)
 
 		// Create a secret that does NOT propogate to bazName
```
Comment (yiqigao217) [inline incubator/hnc/internal/reconcilers/object_test.go:66]: Seems the "secret"s are not replaced by "role"s in the comments.

File: incubator/hnc/internal/reconcilers/test_helpers_test.go (hunk 2 above)
Comment (GinnyJI) [reply incubator/hnc/internal/reconcilers/test_helpers_test.go:200]: As discussed offline we decided to keep it as it is

File: incubator/hnc/internal/reconcilers/object_test.go (hunk 4 above)
Comment (GinnyJI) [reply incubator/hnc/internal/reconcilers/object_test.go]: It should be updated now

**Pull request merged**